packages/flutter/lib/src/material/about.dart

```diff
@@ -1173,21 +1160,14 @@ class _MasterDetailFlowState extends State<_MasterDetailFlow> implements _PageOp
 
   @override
   Widget build(BuildContext context) {
-    switch (widget.displayMode) {
-      case _LayoutMode.nested:
-        return _nestedUI(context);
-      case _LayoutMode.lateral:
+    return LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
+      final double availableWidth = constraints.maxWidth;
+      if (availableWidth >= _materialWideDisplayThreshold) {
         return _lateralUI(context);
-      case _LayoutMode.auto:
-        return LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
-          final double availableWidth = constraints.maxWidth;
-          if (availableWidth >= _materialWideDisplayThreshold) {
-            return _lateralUI(context);
-          } else {
-            return _nestedUI(context);
-          }
-        });
-    }
+      } else {
+        return _nestedUI(context);
+      }
+    });
```

Comment:
nit: avoid else after return

Reply:
Good call. We should probably have a lint for that...

Reply:
https://github.com/dart-lang/linter/issues/4363